packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -407,6 +416,16 @@ enum TargetPlatform {
   android_x86,
 }
 
+enum WindowsArch {
```

Comment:
Should these not be target platforms? for reference TODO on 431.

Reply:
Actually going the exact opposite direction, target platforms are already super overloaded. Rather than say "hey we're building for ios_armA, ios_armB" we say "hey we're building for ios, with architectures armA, armB"

Reply:
Ok, that makes sense. I'm thinking that, in terms of the cache refactor, I should re-write these as classes then, so that we can model the hierarchical relationship.

Reply:
I think for caching we can flatten them back out to strings. But we should update the cache to distinguish between HostPlatform and TargetPlatform if possible